documentation/design/proposals/authn_authz_for_1.0.adoc

```diff
@@ -0,0 +1,329 @@
+      credentials: # borrow structure from connector
+    internal:
+      enabled: boolean
+    clientAuth:
+      enabled: boolean
+    serviceAccount:
+       enabled: boolean
+    anonymous:
+       enabled: boolean
+
+  # Allows specifying a selector for namespaces that should use this accesscontrol by default.
+  namespaceSelector:
+    matchLabels:
+      type: messaging
+status:
+  phase: Active
+```
+
+==== MessagingInfra
+
+Configuration provided by `MessagingInfra` will define the Access Control Server deployment.
+
+```
+accessControl:
+    replicas: <integer>
+```
+
+==== MessagingTennant
+
+A `MessagingTennant` can explictly reference a AccessControlService in order to override default configuration provided by the
+Service Admin.
+
+```
+accessControlService:
+    name: myaccesscontrol1
+    namespace: custom
+```
+
+==== MessagingUser
+
+`MessagingUser` exists to supports the internal identity source.  It allows a Tenant Admin to define a user/password that
+may connect for messaging.
+
+The stored password will be the Salted Password defined by  https://tools.ietf.org/html/rfc5802. 
+The controller will detect newly created MessagingUser resource and automatically salt plain text passwords.
+
+In this first version, there will be no provision for password policies (constitunency rules, aging etc).
+
+```
+apiVersion: enmasse.io/v1
+kind: MessagingUser
+metadata:
+  name: myuser
+  namespace: mynamespace
+spec:
+  password: {SCRAMSHA256}_saltedpassword_
+```
+
+==== ServiceAccount
+
+The bearer token of a `ServiceAccounts` can be used to authenticate to the namespace for messaging.  The service accoun
+must exist within the namespace of the messaging tenant.
+
+==== MessagingRole
+
+`MessagingRole` is used with `MessagingRoleBinding` to define authorisation rules.  The `MessagingRole` allows
+one or more address patterns to the associated with the verbs "send" and/or "receive".   Permissions are purely additive (there are no “deny” rules).  There is no referential integrity enforced between the address prefixes and the
+messaging addresses.
+
+```
+apiVersion: enmasse.io/v1
+kind: MessagingRole
+metadata:
+  name: myrole
+  namespace: mynamespace
+spec:
+- addressPatterns: ["alpha*", "beta", "gamma"]
+  verbs: ["send", "receive"]
+```
+
+==== MessagingRoleBinding
+
+The `MessagingRoleBinding` provides the linkage between the user and the `MessagingRole`. There is no referential integrity enforced between the referenced messaging users or service accounts and the object themselves.
+
+```
+apiVersion: enmasse.io/v1
+kind: MessagingRoleBinding
+metadata:
+  name: myapps1-permissions
+  namespace: mynamespace
+subjects:
+- kind: MessagingUser
+  name: myuser  # refers to a MessagingUser
+- kind: ServiceAccount
+  name: myapp  # refers to a ServiceAccount  
+- kind: Subject
+  name: "foouser@ldap.mycompany.com". # refers to an external subject which must be domain qualified.
```

Comment:
```suggestion
  name: "foouser@ldap.mycompany.com". # refers to an external subject which must be domain qualified.   This can refer to either user or group subjects.
```